File: scrimage-core/src/main/scala/com/sksamuel/scrimage/canvas/drawable/Rect.scala
package com.sksamuel.scrimage.canvas.drawable

import java.awt.Graphics2D

import com.sksamuel.scrimage.canvas.{Drawable, GraphicsContext}

case class Rect(x: Int, y: Int, width: Int, height: Int, context: GraphicsContext) extends Drawable {

  def draw(g: Graphics2D): Unit = g.drawRect(x, y, width, height)

  def fill: FilledRect = FilledRect(x, y, width, height, context)

  def rounded(arcWidth: Int, arcHeight: Int): RoundedRect =
    RoundedRect(x, y, width, height, arcWidth, arcHeight, context)
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/Rect.java
package com.sksamuel.scrimage.canvas.drawables;

import com.sksamuel.scrimage.canvas.Drawable;
import com.sksamuel.scrimage.canvas.GraphicsContext;
import com.sksamuel.scrimage.canvas.RichGraphics2D;

public class Rect implements Drawable {

   private final int x;
   private final int y;
   private final int width;
   private final int height;
   private final GraphicsContext context;

   public Rect(int x, int y, int width, int height) {
      this(x, y, width, height, GraphicsContext.identity());
   }

   public Rect(int x, int y, int width, int height, GraphicsContext context) {
      this.x = x;
      this.y = y;
      this.width = width;
      this.height = height;
      this.context = context;
   }


   @Override
   public void draw(RichGraphics2D g) {
      g.drawRect(x, y, width, height);
   }

   public FilledRect toFilled() {
      return new FilledRect(x, y, width, height, context);
   }

   @Override
   public GraphicsContext context() {
      return context;
   }
}
